src/sugar3/graphics/progressicon.py

```diff
@@ -14,86 +14,171 @@
 # License along with this library; if not, write to the
 # Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 # Boston, MA 02111-1307, USA.
+'''
+The progressicon module provide a progress icon; a widget like progress
+bar which shows progress of a task 
+'''
 
 from gi.repository import Gtk
 from sugar3.graphics.icon import get_surface
 from sugar3.graphics import style
-
-
-class ProgressIcon(Gtk.DrawingArea):
-    """Display the progress filling the icon.
+ 
+class ProgressIcon(Gtk.DrawingArea): '''UI interface for Progress Icon.'''
+ '''Display the progress filling the icon.
 
     This class is compatible with the sugar3.graphics.icon.Icon class.
 
     Call update(progress) with the new progress to update the icon.
-
     The direction defaults to 'vertical', in which case the icon is
     filled from bottom to top.  If direction is set to 'horizontal',
     it will be filled from right to left or from left to right,
     depending on the system's language RTL setting.
 
-    """
-    def __init__(self, icon_name, pixel_size, stroke_color, fill_color,
+ '''
+    def __init__(self, icon_name, pixel_size, stroke_color, fill_color, 
                  direction='vertical'):
-        Gtk.DrawingArea.__init__(self)
-
-        self._icon_name = icon_name
-        self._direction = direction
-        self._progress = 0
+        ''' 
+        Initializes the ProgressIcon class
+        
+        Parameters:
+           icon_name - The name of icon progressicon uses.
+           pixel_size - It explains basic geometry of an image. 
+           stroke_color - Stroke color means border color. Here, it is for progressicon.
+           fill_color - The inside color of progressicon. 
+        
+        Example :
+          
+           class Box:
+           def area(self):
+           return self.width * self.height
+
+           def __init__(self, width, height):
+           self.width = width
+           self.height = height
+
+           # Create an instance of Box.
+           x = Box(10, 2)
+
+           # Print area.
+           print(x.area())
+        
+        '''
```

Comment:
I don't understand why this example would be documented for this class or this function. I think the explanation of the parameters is sufficient here.